### PR TITLE
Update core,client-hc and reader-osm to Java 8

### PR DESCRIPTION
**Pull request by easbar**
Now that it has been decided that GraphHopper only fully supports Android up to GraphHopper version 1.0 we can update all modules to Java 8. Related: #1940 (I.e. GraphHopper 2.0 drops support for very old Android versions for offline routing)

### CODE REVIEW

File: client-hc/pom.xml [hunk 1]
```diff
@@ -54,19 +54,4 @@
             <scope>test</scope>
         </dependency>
     </dependencies>
-    <build>
-        <plugins>
-            <plugin>
-                <groupId>org.apache.maven.plugins</groupId>
-                <artifactId>maven-compiler-plugin</artifactId>
-                <configuration>
-                    <compilerArgument>-XDignore.symbol.file</compilerArgument>
```
Comment (easbar) [inline client-hc/pom.xml:63]: Does anyone know what this `ignore.symbol.file` was meant to do and if it is still needed?

**Comment by easbar** [comment]
Travis build (but not the one for jdk 8) fails with 

```
$ wget https://raw.githubusercontent.com/sormuras/bach/master/install-jdk.sh
$ . ./install-jdk.sh -F 14 -C
install-jdk.sh 2020-06-02
curl: (52) Empty reply from server
```

File: client-hc/pom.xml (hunk 1 above)
Comment (otbutz) [reply client-hc/pom.xml:63]: Looks like the intention was access to internal JRE classes.

**Comment by otbutz** [comment]
A bit offtopic but we could switch to OpenJDK and retrieve the JDKs from https://adoptopenjdk.net/

**Comment by karussell** [comment]
> A bit offtopic but we could switch to OpenJDK and retrieve the JDKs from https://adoptopenjdk.net/

The install script already fetches the files for OpenJDK:

> install-jdk.sh main purpose is to install the latest-and-greatest available OpenJDK release from jdk.java.net.

**Comment by karussell** [comment]
@otbutz have changed the download servers to adoptopenjdk as java.net has some problems it seems.

**Pull request merged**